name-registry/src/interface.sw

```diff
@@ -0,0 +1,99 @@
+library interface;
+
+abi NameRegistry {
+    /// Extends the duration of ownership for the name
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - The name to extend the duration of
+    /// * `duration` - The duration to extend by
+    ///
+    /// # Reverts
+    ///
+    /// * If the name is not registered
+    /// * If the payment is not sufficient for the duration
+    /// * If the wrong asset is sent
+    #[storage(read, write)]
+    fn extend(name: str[8], duration: u64);
+
+    /// Registers an entry into the registry for the given name, assigns it to the given owner, and resolves the name to the given identity when queried
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - The name to register
+    /// * `duration` - The duration to register for
+    /// * `owner` - The owner of the name, which will be able to control the ownership and the resolving identity of the name
+    /// * `identity` - The identity to which the name would resolve to when queried
+    ///
+    /// # Reverts
+    ///
+    /// * If the name is registered or if it has not expired
+    /// * If the payment is not sufficient for the duration
+    /// * If the wrong asset is sent
+    #[storage(read, write)]
+    fn register(name: str[8], duration: u64, owner: Identity, identity: Identity);
+
+    /// Sets the identity to which the name will resolve to
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - The name to set the identity for
+    /// * `identity` - The identity which the name will resolve to
+    ///
+    /// # Reverts
+    ///
+    /// * If the name is registered
```

Comment:
I don't think that this revert statement is correct, including the functions below.